### PR TITLE
Oops! All prosthetics! Adds -6 quirk with all prosthetic limbs + Icon Tutorial

### PR DESCRIPTION
**Pull request by KirbyDaMaster**
## About The Pull Request
This PR adds a new negative quirk. You start the round with all 4 limbs as prosthetics. This is only -8, which is 2x as many negative points as just having one, so its not quite economical, but would definitely leave you screwed after an EMP

I also created a readme tutorial for adding custom SVG icons, with steps and recommended software.

## Why It's Good For The Game
Having the option to become a quad-amputee for a couple extra negative quirk points isnt quite over the top, but gives a little more risk reward to the mix. It opens you up to some really wacky shenanigans with EMPs and the medbay will just love you (not).

Also having a tutorial for SVG icon creation was an absolute necessity.

## Changelog

:cl:
add: -6 Quirk for all prosthetics named "Quadruple Amputee"
change: prosthetic arm changed from -4 to -3
add: SVG icon tutorial readme to tgui tgfont folder
change: Minor spelling issues in negative quirks tab
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
Let me have all basic cyber organs as well. I want to fucking PERISH if EMP'd.

File: tgui/packages/tgfont/README_ICON_TUTORIAL.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+1. Get whatever SVG you plan on using and put it in the ..\tgstation\tgui\packages\tgfont\icons folder.
+
+2. In VScode, at the top, open "Terminal>New Terminal" and run this exact line of code without the quotes -> ".\BUILD.cmd tg-font"
+    Wait till its all complete
+
+3. In ..\tgstation\tgui\packages\tgfont\dist you will find 3 files that were generated, a .css, .eot and a .WOFF2 file. Copy and replace all 3 of these files to
+    the following directory -> ..\tgstation\tgui\packages\tgfont\static
+
+4. Then we run/double click the BUILD.bat file in ..\tgstation\
```
Comment (Mothblocks) [inline tgui/packages/tgfont/README_ICON_TUTORIAL.txt:13]: wait why are you doing any of this. you should be able to just build. if you cant that is a bug

**Comment by Mothblocks** [comment]
the writing on that guide is fairly sloppy but im also not sure its necessary, i think you were just overcomplicating your workflow

File: code/datums/quirks/negative.dm [hunk 2]
```diff
@@ -455,6 +457,47 @@
...
+	prostheticLA = new/obj/item/bodypart/l_arm/robot/surplus(quirk_holder)
+	prostheticRA = new/obj/item/bodypart/r_arm/robot/surplus(quirk_holder)
+	prostheticLL = new/obj/item/bodypart/l_leg/robot/surplus(quirk_holder)
+	prostheticRL = new/obj/item/bodypart/r_leg/robot/surplus(quirk_holder)
+
+	prostheticLA.replace_limb(human_holder)
+	prostheticRA.replace_limb(human_holder)
+	prostheticLL.replace_limb(human_holder)
+	prostheticRL.replace_limb(human_holder)
+	qdel(old_LA, old_RA, old_LL, old_RL)
```
Comment (Time-Green) [inline code/datums/quirks/negative.dm:494]: This is not ideal. Could you just make a list with the new types and use a for loop to replace the limbs? It’s also not necessary to store and delete all the old limbs, replace_limb probably has a paramater to just delete the replaced limb

File: tgui/packages/tgfont/static/tgfont.css [hunk 3]
```diff
@@ -1,7 +1,7 @@
 @font-face {
     font-family: "tgfont";
-    src: url("./tgfont.woff2?45c3c7acc69dd413375d77898d24e41e") format("woff2"),
```
Comment (Time-Green) [inline tgui/packages/tgfont/static/tgfont.css:3]: What are you doing with the fonts?

File: tgui/packages/tgfont/README_ICON_TUTORIAL.txt [hunk 4]
```diff
@@ -0,0 +1,26 @@
+The following is the process to implement your own icon using an svg.
+
+If you plan on making your own SVG, i suggest Inkscape, as it is free and pretty powerful for vector graphics.
+
+1. Get whatever SVG you plan on using and put it in the ..\tgstation\tgui\packages\tgfont\icons folder.
```
Comment (Time-Green) [inline tgui/packages/tgfont/README_ICON_TUTORIAL.txt:5]: You should atomize your PR’s, this is not very relevant to adding the quirk

File: .gitignore [hunk 5]
```diff
@@ -1,209 +0,0 @@
...
-# Built auxtools libraries and intermediate files
-aux*.dll
-libaux*.so
-aux*.pdb
-
-# JavaScript tools
-**/node_modules
-
-# Screenshot tests
-/artifacts
```
Comment (Time-Green) [inline .gitignore:209]: Uh please don’t delete the gitignore

**Review by Time-Green — CHANGES_REQUESTED** [review]

**Comment by ShizCalev** [comment]
Atomize.

File: code/datums/quirks/negative.dm [hunk 6]
```diff
@@ -417,9 +417,11 @@
 	icon = "user-secret"
 	value = -4
 	mob_trait = TRAIT_PROSOPAGNOSIA
-	medical_record_text = "Patient suffers from prosopagnosia and cannot recognize faces."
+	medical_record_text = "Patient suffers from Prosopagnosia and cannot recognize faces."
```
Comment (tralezab) [inline code/datums/quirks/negative.dm:420]: this is not correct, prosopagnosia shouldn't be capitalized here

File: code/datums/quirks/negative.dm [hunk 7]
```diff
@@ -98,7 +98,7 @@
 	 */
 /datum/quirk/item_quirk/brainproblems
 	name = "Brain Tumor"
-	desc = "You have a little friend in your brain that is slowly destroying it. Better bring some mannitol!"
+	desc = "You have a little friend in your brain that is slowly destroying it. Better bring some Mannitol!"
```
Comment (tralezab) [inline code/datums/quirks/negative.dm:101]: again, no capitalization needed

File: tgui/packages/tgfont/README_ICON_TUTORIAL.txt [hunk 8]
```diff
@@ -0,0 +1,26 @@
+The following is the process to implement your own icon using an svg.
```
Comment (tralezab) [inline tgui/packages/tgfont/README_ICON_TUTORIAL.txt:1]: What everyone else has been saying, atomize this out. Means move it to a different pr

**Review by tralezab — CHANGES_REQUESTED** [review]

**Comment by VaelophisNyx** [comment]
yog coder here to suggest the alternative: just make each limb selectable for -2 with no incompatibilities outside the random 1 limb quirk like we did over on yog?

**Comment by KirbyDaMaster** [comment]
> Why add the icon tutorial in a completely unrelated pr?

It absolutely is required, you cannot add a custom icon for the quirk without following those exact steps. It was not described elsewhere.

File: tgui/packages/tgfont/README_ICON_TUTORIAL.txt (hunk 4 above)
Comment (KirbyDaMaster) [reply tgui/packages/tgfont/README_ICON_TUTORIAL.txt:5]: > You should atomize your PR’s, this is not very relevant to adding the quirk

Absolutely is relevant because if anyone wants to make a quirk in the future this is the only way to add a custom icon to the quirk.

File: tgui/packages/tgfont/static/tgfont.css (hunk 3 above)
Comment (KirbyDaMaster) [reply tgui/packages/tgfont/static/tgfont.css:3]: Its the only way to implement the icon. It is the method outlined by Anturk.

File: tgui/packages/tgfont/README_ICON_TUTORIAL.txt (hunk 1 above)
Comment (KirbyDaMaster) [reply tgui/packages/tgfont/README_ICON_TUTORIAL.txt:13]: > wait why are you doing any of this. you should be able to just build. if you cant that is a bug

Ask Anturk, hes the one managing the system and I followed his steps.

File: code/datums/quirks/negative.dm (hunk 2 above)
Comment (DaedalusGame) [reply code/datums/quirks/negative.dm:494]: https://github.com/tgstation/tgstation/blob/master/code/modules/surgery/bodyparts/dismemberment.dm#L316 no parameter to delete the replaced limb, hence the existing quirk doing the same thing

**Comment by tralezab** [comment]
you know what if kirby says it's related and required I'll keep it in

File: code/datums/quirks/negative.dm (hunk 7 above)
Comment (Iamgoofball) [reply code/datums/quirks/negative.dm:101]: but it's referring to a proper noun, Mannitol

Comment (Time-Green) [reply code/datums/quirks/negative.dm:101]: Medicines are not proper nouns, they’re just trivial names to easily refer to compounds. It’s not a person, place or thing

**Comment by optimumtact** [comment]
who the hell are these atomizing andy's in here? the readme is directly relevant and is due to an undocumented process that is required.

adding it in this PR is fine?

**Comment by KirbyDaMaster** [comment]
> the writing on that guide is fairly sloppy but im also not sure its necessary, i think you were just overcomplicating your workflow

Sloppy how? please elaborate.

File: code/datums/quirks/negative.dm (hunk 2 above)
Comment (Time-Green) [reply code/datums/quirks/negative.dm:494]: That code could also be better, but I think it’s a bit cruel to suggest an associative static list. I think doing what you suggested is good for now

File: .gitignore (hunk 5 above)
Comment (stylemistake) [reply .gitignore:209]: what the fuck

File: tgui/packages/tgfont/static/tgfont.css (hunk 3 above)
Comment (stylemistake) [reply tgui/packages/tgfont/static/tgfont.css:3]: when new icons are added, rebuild of fonts and css is required, which changes the hash of the font file

File: tgui/packages/tgfont/README_ICON_TUTORIAL.txt (hunk 1 above)
Comment (Mothblocks) [reply tgui/packages/tgfont/README_ICON_TUTORIAL.txt:13]: @AnturK theres no way you need to copy and replace files for tgfont bro

File: code/datums/quirks/negative.dm [hunk 9]
```diff
@@ -447,14 +449,37 @@
...
+
+/datum/quirk/quadruple_amputee/add_unique()
+
+	var/mob/living/carbon/human/human_holder = quirk_holder
+	var/list/zones = list(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+	var/list/prosthetics = list(/obj/item/bodypart/l_arm/robot/surplus, /obj/item/bodypart/r_arm/robot/surplus, /obj/item/bodypart/l_leg/robot/surplus, /obj/item/bodypart/r_leg/robot/surplus)
+	for(var/i in 1 to 4)
+		var/path = prosthetics[i]
+		var/obj/item/bodypart/prosthetic = new path
+		var/obj/item/bodypart/old_limbs = human_holder.get_bodypart(zones[i])
```
Comment (Mothblocks) [inline code/datums/quirks/negative.dm:475]: loop over zones directly instead

File: code/datums/quirks/negative.dm [hunk 10]
```diff
@@ -447,14 +449,37 @@
...
+		prosthetic.replace_limb(human_holder, special = TRUE)
+		qdel(old_limbs)
+
+/datum/quirk/quadruple_amputee/post_add()
+	to_chat(quirk_holder, "<span class='boldannounce'>All your limbs have been replaced with surplus prosthetics. They are fragile and will easily come apart under duress. Additionally, \
```
Comment (Mothblocks) [inline code/datums/quirks/negative.dm:480]: use `span_boldannounce()`

File: code/datums/quirks/negative.dm [hunk 11]
```diff
@@ -447,14 +449,37 @@
...
 /datum/quirk/prosthetic_limb/post_add()
 	to_chat(quirk_holder, "<span class='boldannounce'>Your [slot_string] has been replaced with a surplus prosthetic. It is fragile and will easily come apart under duress. Additionally, \
 	you need to use a welding tool and cables to repair it, instead of bruise packs and ointment.</span>")
 
+/datum/quirk/quadruple_amputee
+	name = "Quadruple Amputee"
+	desc = "Oops! All Prosthetics! Due to some truly cruel cosmic punishment, all your limbs have been taken from you."
+	icon = "tg-prosthetic-full"
+	value = -8
+	medical_record_text = "During physical examination, patient was found to have all prosthetic limbs"
```
Comment (Mickyan) [inline code/datums/quirks/negative.dm:464]: Missing period at the end

**Comment by Mickyan** [comment]
Not too fond of this because prosthetic limbs can almost be considered sidegrades to regular limbs (no wounds, easy to repair trading for EMP weakness and lower max health) and the prosthetic limb quirk is one of those most people already consider free points

Personally I would lower the point value to -6 or so, but this definitely needs to be blacklisted from Frail at the very least

**Comment by RuneQuester** [comment]
SO CLOSE to being able to choose WHICH limb is prosthetic. we are SO close

File: code/datums/quirks/negative.dm [hunk 12]
```diff
@@ -447,14 +449,37 @@
...
+
+	var/mob/living/carbon/human/human_holder = quirk_holder
+	var/list/zones = list(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+	var/list/prosthetics = list(/obj/item/bodypart/l_arm/robot/surplus, /obj/item/bodypart/r_arm/robot/surplus, /obj/item/bodypart/l_leg/robot/surplus, /obj/item/bodypart/r_leg/robot/surplus)
+	for(var/i in 1 to 4)
+		var/path = prosthetics[i]
+		var/obj/item/bodypart/prosthetic = new path
+		var/obj/item/bodypart/old_limbs = human_holder.get_bodypart(zones[i])
+		prosthetic.replace_limb(human_holder, special = TRUE)
+		qdel(old_limbs)
```
[the remark this answers is not in the document]
Comment (DaedalusGame) [reply code/datums/quirks/negative.dm:477]: I don't see this at all chief

- replace_body_part isn't an existing proc, replace_limb is but doesn't qdel the old limb
- if it doesn't matter whether the old limb is qdel'd we should just use replace_limb instead
- otherwise you're asking for a new proc to be created that will be confusing to people getting started with limb code and will only be used in like 1 or 2 places

**Comment by KirbyDaMaster** [comment]
> Not too fond of this because prosthetic limbs can almost be considered sidegrades to regular limbs (no wounds, easy to repair trading for EMP weakness and lower max health) and the prosthetic limb quirk is one of those most people already consider free points
> 
> Personally I would lower the point value to -6 or so, but this definitely needs to be blacklisted from Frail at the very least

resolved

File: code/datums/quirks/negative.dm (hunk 11 above)
Comment (KirbyDaMaster) [reply code/datums/quirks/negative.dm]: changed

File: code/datums/quirks/negative.dm (hunk 10 above)
Comment (KirbyDaMaster) [reply code/datums/quirks/negative.dm]: made the change

File: code/datums/quirks/negative.dm (hunk 12 above)
[the remark this answers is not in the document]
Comment (KirbyDaMaster) [reply code/datums/quirks/negative.dm]: updated

File: code/__HELPERS/bodyparts.dm [hunk 13]
```diff
@@ -1 +1,6 @@
 #define IS_ORGANIC_LIMB(limb) (limb.bodytype & BODYTYPE_ORGANIC)
+
+/proc/replace_body_part(mob/living/carbon/human/human_holder, zone, obj/item/bodypart/prosthetic)
+	var/obj/item/bodypart/old_limb = human_holder.get_bodypart(zone)
+	prosthetic.replace_limb(human_holder, special = TRUE)
+	qdel(old_limb)
```
Comment (Kapu1178) [inline code/__HELPERS/bodyparts.dm:6]:
```suggestion
/mob/living/carbon/proc/replace_body_part(mob/living/carbon/human/human_holder, zone, obj/item/bodypart/prosthetic)
	var/obj/item/bodypart/old_limb = human_holder.get_bodypart(zone)
	prosthetic.replace_limb(human_holder, special = TRUE)
	qdel(old_limb)
```

Comment (DaedalusGame) [reply code/__HELPERS/bodyparts.dm:6]: cool but you forgot to change the holder from parameter to src

**Comment by JohnFulpWillard** [comment]
Please handle @Kapu1178 's reviews

File: code/__HELPERS/bodyparts.dm (hunk 13 above)
Comment (JohnFulpWillard) [reply code/__HELPERS/bodyparts.dm:6]: ``src.get_bodypart``
💀

**Review by KirbyDaMaster — COMMENTED** [review]
I think what is here is fine.

File: code/__HELPERS/bodyparts.dm [hunk 14]
```diff
@@ -1 +1,7 @@
 #define IS_ORGANIC_LIMB(limb) (limb.bodytype & BODYTYPE_ORGANIC)
+
+/mob/living/carbon/proc/del_and_replace_bodypart(obj/item/bodypart/new_limb, special)
```
Comment (Mothblocks) [inline code/__HELPERS/bodyparts.dm:3]: 1. Document this
2. Put this in the mob folder, __HELPERS should be kept somewhat slim/global

Comment (DaedalusGame) [reply code/__HELPERS/bodyparts.dm:3]: Document where

Comment (Mothblocks) [reply code/__HELPERS/bodyparts.dm:3]: A triple slashed comment makes a doc comment

```
/// Doc comment goes here
/thing/proc/action()
```

code/modules/mob/living/carbon, and make like carbon_bodyparts.dm

Comment (DaedalusGame) [reply code/__HELPERS/bodyparts.dm:3]: why not put it in dismember code

Comment (Mothblocks) [reply code/__HELPERS/bodyparts.dm:3]: Sure just not in a helper

Comment (DaedalusGame) [reply code/__HELPERS/bodyparts.dm]: Done

File: tgui/packages/tgfont/README_ICON_TUTORIAL.txt [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+NOTE; Be sure when you reference your icon that you prefix it with "tg-", otherwise it will not find it.
+
+EXAMPLE;
+    for svg named "prosthetic-leg.svg", you would reference it like so;
+        vvvvvvvvvvvvvvvvvvvvvvvvvv
+        icon = "tg-prosthetic-leg"
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^
+    The process uses the main filename of your SVG as the icon name, so don't name it something stupid.
+
+    Also try to get your SVG as simple as possible, the engine has trouble rendering SVGs that have a lot of little disconnected parts
```
Comment (Mothblocks) [inline tgui/packages/tgfont/README_ICON_TUTORIAL.txt:24]:
```suggestion
If you plan on making your own SVG, consider [Inkscape](https://inkscape.org/). It is free and pretty powerful for vector graphics.

1. Get whatever SVG you plan on using and put it in the `tgstation\tgui\packages\tgfont\icons` folder.

2. In VS Code, press Ctrl+Shift+B, and select "tgui: rebuild tgfont". Wait for it to comlpete.

Now your SVG will be able to be used in the game.

When you reference your icon that you prefix it with "tg-", otherwise it will not find it. For example, with an SVG named "prosthetic-leg.svg", you would reference it with `icon_state = "tg-prosthetic-leg"`.

Keep your SVG as simple as possible, the engine has trouble rendering SVGs that have a lot of little disconnected parts.
```

CC @AnturK to make sure I have the new instructions right.

General writing touch ups.

**Review by KirbyDaMaster — COMMENTED** [review]
Outdated

File: code/datums/quirks/negative.dm [hunk 16]
```diff
@@ -447,13 +448,30 @@
...
+	icon = "tg-prosthetic-full"
+	value = -6
+	medical_record_text = "During physical examination, patient was found to have all prosthetic limbs."
+	hardcore_value = 6
+
+/datum/quirk/quadruple_amputee/add_unique()
+	var/mob/living/carbon/human/human_holder = quirk_holder
+	human_holder.del_and_replace_bodypart(BODY_ZONE_L_ARM, new/obj/item/bodypart/l_arm/robot/surplus(quirk_holder))
+	human_holder.del_and_replace_bodypart(BODY_ZONE_R_ARM, new/obj/item/bodypart/r_arm/robot/surplus(quirk_holder))
+	human_holder.del_and_replace_bodypart(BODY_ZONE_L_LEG, new/obj/item/bodypart/l_leg/robot/surplus(quirk_holder))
```
Comment (Time-Green) [inline code/datums/quirks/negative.dm:469]: Params are (/obj/blabla/bodypart, special), this wont work

File: code/datums/quirks/negative.dm [hunk 17]
```diff
@@ -447,13 +448,30 @@
 		if(BODY_ZONE_R_LEG)
 			prosthetic = new/obj/item/bodypart/r_leg/robot/surplus(quirk_holder)
 			slot_string = "right leg"
-	prosthetic.replace_limb(human_holder)
-	qdel(old_part)
-	human_holder.regenerate_icons()
+	human_holder.del_and_replace_bodypart(limb_slot, prosthetic)
```
Comment (Time-Green) [inline code/datums/quirks/negative.dm:451]: Just put this in the switches tbh

**Comment by Time-Green** [comment]
Posthumous verified

**Comment by Mothblocks** [comment]
the verified is only minor changes that doesnt count

**Comment by Time-Green** [comment]
> the verified is only minor changes that doesnt count

verified is more like a state of mind
